packages/contracts/contracts/test/ValidatorManager.t.sol

```diff
@@ -0,0 +1,793 @@
+
+    event ValidatorUnjailed(address indexed validator);
+
+    event RewardDistributed(
+        address indexed validator,
+        uint128 validatorReward,
+        uint128 baseReward,
+        uint128 boostedReward
+    );
+
+    function _submitL2Output(uint256 l2BlockNumber, bool isPublicRound) private {
+        uint256 outputIndex = oracle.nextOutputIndex();
+        if (!isPublicRound) {
+            vm.prank(valMan.nextValidator());
+        }
+        mockOracle.addOutput(l2BlockNumber);
+        vm.prank(address(oracle));
+        valMan.afterSubmitL2Output(outputIndex);
+    }
+
+    function _setUpHundredKghDelegation(address validator, uint256 startingTokenId) private {
+        uint256[] memory tokenIds = new uint256[](100);
+        for (uint256 i = startingTokenId; i < 100 + startingTokenId; i++) {
+            kgh.mint(validator, i);
+            vm.prank(validator);
+            kgh.approve(address(assetMan), i);
+            tokenIds[i - startingTokenId] = i;
+        }
+        vm.prank(validator);
+        assetMan.delegateKghBatch(validator, tokenIds);
+    }
+
+    function setUp() public override {
+        super.setUp();
+
+        address oracleAddress = address(oracle);
+        MockL2OutputOracle mockOracleImpl = new MockL2OutputOracle(
+            pool,
+            valMan,
+            address(colosseum),
+            submissionInterval,
+            l2BlockTime,
+            startingBlockNumber,
+            startingTimestamp,
+            finalizationPeriodSeconds
+        );
+        vm.prank(multisig);
+        Proxy(payable(oracleAddress)).upgradeTo(address(mockOracleImpl));
+        mockOracle = MockL2OutputOracle(oracleAddress);
+
+        address valManAddress = address(valMan);
+        MockValidatorManager mockValManImpl = new MockValidatorManager(constructorParams);
+        vm.prank(multisig);
+        Proxy(payable(valManAddress)).upgradeTo(address(mockValManImpl));
+        mockValMan = MockValidatorManager(valManAddress);
+
+        VKRO_PER_KGH = assetMan.VKRO_PER_KGH();
+    }
+
+    function test_constructor_succeeds() external {
+        assertEq(address(valMan.L2_ORACLE()), address(oracle));
+        assertEq(address(valMan.ASSET_MANAGER()), address(assetMan));
+        assertEq(valMan.TRUSTED_VALIDATOR(), trusted);
+        assertEq(valMan.MIN_REGISTER_AMOUNT(), minRegisterAmount);
+        assertEq(valMan.MIN_START_AMOUNT(), minStartAmount);
+        assertEq(valMan.COMMISSION_RATE_MIN_CHANGE_SECONDS(), commissionRateMinChangeSeconds);
+        assertEq(valMan.ROUND_DURATION_SECONDS(), roundDuration);
+        assertEq(valMan.JAIL_PERIOD_SECONDS(), jailPeriodSeconds);
+        assertEq(valMan.JAIL_THRESHOLD(), jailThreshold);
+        assertEq(valMan.MAX_OUTPUT_FINALIZATIONS(), maxOutputFinalizations);
+        assertEq(valMan.BASE_REWARD(), baseReward);
+    }
+
+    function test_constructor_smallMinStartAmount_reverts() external {
```

Comment:
Let's also add revert test when commission rate is greater than denominator.

Reply:
The commission rate is set when validator registers. The test you mean already exists, `test_registerValidator_largeCommissionRate_reverts`.